activerecord/lib/active_record/connection_adapters/postgresql/oid/box.rb

```diff
@@ -0,0 +1,40 @@
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Box < Type::Value # :nodoc:
+          def type
+            :box
+          end
+
+          def serialize(value)
+            if value.is_a?(::Array)
+              "(#{serialize(value[0])},#{serialize(value[1])})"
+            else
+              value
+            end
+          end
+
+          def cast(value)
+            case value
+            when ::String
```

Comment:
I don't think it makes sense to have different data type returned if casting didn't go as expected (all the `else` cases returning `String`). If it's not possible for postgres box to ever be anything than 4 values, don't need to code for it at all.